src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5723,7 +5732,13 @@ private BoundExpression BindUnexpectedComplexElementInitializer(InitializerExpre
             var enumeratorInfo = element.EnumeratorInfoOpt;
             if (enumeratorInfo is null)
             {
-                return element;
+                return element.Update(
+                    BindToNaturalType(element.Expression, diagnostics),
```

Comment:
Is this an error scenario? If so I thought there was a 'BindForErrorRecovery' method we should be using?

Reply:
Yes, this is an error scenario. I couldn't find a `Bind*ForErrorRecovery` method that seemed appropriate here although I might have missed one.